### PR TITLE
Support glob argument with ``--cache-show``

### PR DESCRIPTION
**Pull request by blueyed**
TODO:

- [x] changelog

### CODE REVIEW

File: src/_pytest/cacheprovider.py [hunk 1]
```diff
@@ -282,9 +282,13 @@ def pytest_addoption(parser):
     )
     group.addoption(
         "--cache-show",
-        action="store_true",
+        action="append",
+        nargs="?",
```
Comment (blueyed) [inline src/_pytest/cacheprovider.py]: There might be a better way?

Comment (nicoddemus) [reply src/_pytest/cacheprovider.py]: I've tried a few things (action="store" with default="*") but that didn't work, so I guess this is fine.

Comment (blueyed) [reply src/_pytest/cacheprovider.py]: Ok, borrowed / inspired by pytest-cov's `--cov`.

**Comment by nicoddemus** [comment]
Great work! 👍

**Pull request merged**